packaging/windows/Jenkinsfile.groovy

```diff
@@ -0,0 +1,169 @@
+        )
+    }
+    // Make sure any previous desktop build is deleted
+    bat '''
+        if EXIST src\\github.com\\keybase\\client\\desktop\\release rmdir /q /s src\\github.com\\keybase\\client\\desktop\\release
+        path
+    '''                
+    stage('Build Client') {
+        bat '"%ProgramFiles(x86)%\\Microsoft Visual Studio 14.0\\vc\\bin\\vcvars32.bat" && src\\github.com\\keybase\\client\\packaging\\windows\\build_prerelease.cmd'
+    } 
+    stage('Build UI') {
+        bat 'src\\github.com\\keybase\\client\\packaging\\windows\\buildui.bat'
+    }
+    stage('Build Installer') {
+        bat 'call "%ProgramFiles(x86)%\\Microsoft Visual Studio 14.0\\vc\\bin\\vcvars32.bat" && src\\github.com\\keybase\\client\\packaging\\windows\\doinstaller_wix.cmd'
+        archiveArtifacts 'src\\github.com\\keybase\\client\\packaging\\windows\\${BUILD_TAG}\\*.*'
+    }
+    
+    if (UpdateChannel != "None"){    
+        stage('Publish to S3') {
+            step([
+                $class: 'S3BucketPublisher',
+                dontWaitForConcurrentBuildCompletion: false,
+                entries: [[
+                    bucket: 'prerelease.keybase.io/windows',
+                    excludedFile: '',
+                    flatten: true,
+                    gzipFiles: false,
+                    keepForever: false,
+                    managedArtifacts: false,
+                    noUploadOnFailure: true,
+                    selectedRegion: 'us-east-1',
+                    showDirectlyInBrowser: false,
+                    sourceFile: 'src\\github.com\\keybase\\client\\packaging\\windows\\${BUILD_TAG}\\*.exe',
+                    storageClass: 'STANDARD',
+                    uploadFromSlave: true,
+                    useServerSideEncryption: false
+                ]],
+                profileName: 'keybase',
+                userMetadata: []
+            ])
+            step([
+                $class: 'S3BucketPublisher',
+                dontWaitForConcurrentBuildCompletion: false,
+                entries: [[
+                    bucket: 'prerelease.keybase.io',
+                    excludedFile: '',
+                    flatten: true,
+                    gzipFiles: false,
+                    keepForever: false,
+                    managedArtifacts: false,
+                    noUploadOnFailure: true,
+                    selectedRegion: 'us-east-1',
+                    showDirectlyInBrowser: false,
+                    sourceFile: 'src\\github.com\\keybase\\client\\packaging\\windows\\${BUILD_TAG}\\update-windows-prod-test-v2.json',
+                    storageClass: 'STANDARD',
+                    uploadFromSlave: true,
+                    useServerSideEncryption: false
+                ]],
+            profileName: 'keybase',
+            userMetadata: []])
+        }
+    }
+
+    if (UpdateChannel == "Smoke"){
+        stage('Invoke SmokeB build') {
```

Comment:
Stages should always exist or not exist; branched stages mess up the analytics. So I'd put it above the if, and then have an else clause that echoes that it isn't doing anything.